### ISSUE TITLE
[question]: How to solve this dependency conflict

### ISSUE THREAD

**Issue by Martin-Gonzalez90** [OneSignal/OneSignal-Android-SDK#1628]
### How can we help?

Hi, I'm using the Unity SDK but I'm trying to solve the native side of Android.
When trying to sync gradle I'm having an dependency resolution error with firebase-messaging.

If I run `./gradlew --info assembleDebug`I get:

```
Dependency resolved to an incompatible version: Dependency(fromArtifactVersion=ArtifactVersion(groupId=com.onesignal, artifactId=OneSignal, version=4.8.1), toArtifact=Artifact(groupId=com.google.firebase, artifactId=firebase-messaging), toArtifactVersionString=[19.0.0, 22.0.99])
Dependency Resolution Help: Displaying all currently known paths to any version of the dependency: Artifact(groupId=com.google.firebase, artifactId=firebase-messaging)
NOTE: com.google.android.gms translated to c.g.a.g for brevity. Same for com.google.firebase -> c.g.f
-- :Launchpad-debugCompileClasspath:0.0.0 task/module dep -> com.onesignal:OneSignal@4.8.1
---- com.onesignal:OneSignal:4.8.1 library depends -> c.g.f:firebase-messaging@[19.0.0, 22.0.99]
-- :Launchpad-debugCompileClasspath:0.0.0 task/module dep -> com.onesignal:OneSignal@{strictly 4.8.1}
---- com.onesignal:OneSignal:4.8.1 library depends -> c.g.f:firebase-messaging@[19.0.0, 22.0.99]
-- :Launchpad-debugCompileClasspath:0.0.0 task/module dep -> c.g.f:firebase-messaging@{strictly 20.2.1}
-- :Launchpad-debugCompileClasspath:0.0.0 task/module dep -> c.g.f:firebase-messaging@20.2.1

FAILURE: Build failed with an exception.
```

What I cannot understand is that I'm using a firebase messaging version that is in that range.
Any thoughts?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by brismithers** [comment]
@Martin-Gonzalez90 unfortunately I haven't been able to reproduce your issue.  I was able to specify the below dependencies  and the build process will use `com.google.firebase:firebase-messaging@20.2.1` without issue.

```
    implementation 'com.onesignal:OneSignal:4.8.1'
    implementation('com.google.firebase:firebase-messaging') {
        version {
            strictly '20.2.1'
        }
    }
```

Would you be able to provide the dependency portion of your `build.gradle` ?  If you could trim it down to only the dependencies required to reproduce the issue that would be appreciated, thanks!

**Comment by Martin-Gonzalez90** [comment]
> @Martin-Gonzalez90 unfortunately I haven't been able to reproduce your issue. I was able to specify the below dependencies and the build process will use `com.google.firebase:firebase-messaging@20.2.1` without issue.
> 
> ```
>     implementation 'com.onesignal:OneSignal:4.8.1'
>     implementation('com.google.firebase:firebase-messaging') {
>         version {
>             strictly '20.2.1'
>         }
>     }
> ```
> 
> Would you be able to provide the dependency portion of your `build.gradle` ? If you could trim it down to only the dependencies required to reproduce the issue that would be appreciated, thanks!

Thanks for reaching out.
I was able to fix this by using gradle plugin 4.0.0. Basically by doing that and changing the gradle wrapper version to 6.1.1 I was able to build. To be honest I'm not advance in android native, do you know why this could be the fix?
If I take out OneSignal I can use back a previous gradle version
